AppInsightsWithWebJob/Program.cs

```diff
@@ -37,14 +38,13 @@ static async Task Main(string[] args)
             var host = builder.Build();
             var cancellationToken = new WebJobsShutdownWatcher().Token;
 
-            var telemetryClient = host.Services.GetService<TelemetryClient>();
-            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.ExceptionObject);
-            TaskScheduler.UnobservedTaskException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.Exception);
+//            var telemetryClient = host.Services.GetService<TelemetryClient>();
+//            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.ExceptionObject);
+//            TaskScheduler.UnobservedTaskException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.Exception);
 
-            using (cancellationToken.Register(() => host.Services.GetService<IJobHost>().StopAsync().GetAwaiter().GetResult()))
             using (host)
             {
-                await host.RunAsync(cancellationToken);
+                await host.RunAsync(default(CancellationToken));
```

Comment:
Remove the argument from RunAsync there's an overload that doesnt take an argument.